src/common/snippets/src/lowered/linear_ir.cpp

```diff
@@ -43,8 +44,11 @@ LinearIR::LinearIR(const std::shared_ptr<ov::Model>& model,
             insertion_pos = std::next(last_param);
         }
 
-        // exec_num = 0 since `insertion_pos` can be changed
-        register_expression(expr, true, 0);
+        // Some utils containers (for example, buffers) in Lir contain expressions in execution order
+        // so we have to pass exec order to registration. However, this enumeration is not optimal because
+        // the next each expr will has exec_num = prev_expr->exec_num + 1.
```

Comment:
I didn't get it. Why the next expr will have and exec number incremented by 1 when you pass the result of `get_inserted_expr_exec_num(insertion_pos)` to `register_expression`? Is it because we always insert the last expression when creating the LIR?
If so, we shouldn't bother calling the `get_inserted_expr_exec_num` here. Let's just pass 0 to make it clear that the enumeration is irrelevant at this point. We will call enumerate on L61 anyway

Reply:
Discussed offline: only `SetBufferRegGroup` requires ordered buffers -> now buffer container stores unsorted expressions but we locally sort them only in the needed pass. Thanks!